app/stable_appconfig.go

```diff
@@ -59,7 +68,88 @@ func setCustomOrderEndBlocker() []string {
 	return []string{}
 }
 
-func (app *RegenApp) registerUpgradeHandlers() {}
+func (app *RegenApp) registerUpgradeHandlers() {
+
+	// mainnet upgrade handler
+	const upgradeName = "v4.0.0"
+	app.UpgradeKeeper.SetUpgradeHandler(upgradeName, func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		// run state migrations for sdk modules
+		toVersion, err := app.mm.RunMigrations(ctx, app.configurator, fromVM)
+		if err != nil {
+			return nil, err
+		}
+
+		// run x/ecocredit state migrations
+		if err := app.smm.RunMigrations(ctx, app.AppCodec()); err != nil {
+			return nil, err
+		}
+		toVersion[ecocredit.ModuleName] = ecocreditmodule.Module{}.ConsensusVersion()
+
+		// recover funds for community member (regen-1 governance proposal #11)
+		if ctx.ChainID() == "regen-1" {
+			if err := recoverFunds(ctx, app.AccountKeeper, app.BankKeeper); err != nil {
+				return nil, err
+			}
+		}
+
+		return toVersion, nil
+	})
+
+}
+
+func recoverFunds(ctx sdk.Context, ak authkeeper.AccountKeeper, bk bankkeeper.Keeper) error {
+	// address with funds inaccessible
+	const lostAddr = "regen1c3lpjaq0ytdtsrnjqzmtj3hceavl8fe2vtkj7f"
+	// address that the community member has access to
+	const newAddr = "regen14tpuqrwf95evu3ejm9z7dn20ttcyzqy3jjpfv4"
+
+	var lostAccount, newAccount authtypes.AccountI
+	ak.IterateAccounts(ctx, func(account authtypes.AccountI) (stop bool) {
+		addr := account.GetAddress().String()
+		if addr == lostAddr {
+			lostAccount = account
+			if newAccount != nil {
+				return true
+			}
+		} else if addr == newAddr {
+			newAccount = account
+			if lostAccount != nil {
+				return true
+			}
+		}
+
+		return false
+	})
```

Comment:
cant we simply use 

```
lostAccout := regenApp.AccountKeeper. GetAccount(ctx, lostAddr)
newAccount := regenApp.AccountKeeper. GetAccount(ctx, lostAddr)
```

Also seems like this doesn't work if `newAccount` is nil